packages/block-library/src/navigation/edit/inner-blocks.js

```diff
@@ -88,6 +89,11 @@ export default function NavigationInnerBlocks( {
 	const showPlaceholder =
 		! hasCustomPlaceholder && ! hasMenuItems && ! isSelected;
 
+	// If the `defaultBlock` attribute is set and itself has an attributes object with values then use it
+	// otherwise fallback to the DEFAULT_BLOCK constant.
+	// This allows site owners to set the default `core/navigation-link` variation to use.
+	const navDefaultBlock = defaultBlock && defaultBlock.attributes && Object.keys(defaultBlock.attributes).length > 0 ? defaultBlock : DEFAULT_BLOCK;
```

Comment:
I'm not sure about the part that checks the length of the attributes. Wouldn't that disallow setting a new default that doesn't have any specific default attributes?

Reply:
@talldan Thats correct, my intention here was to assume if a developer/user hasn't included some attributes that actually changes something about the default `core/navigation-link` block to be inserted that we'd default to DEFAULT_BLOCK. But I can also see how that might confuse people at first. Ideally I think we'd get rid of `DEFAULT_BLOCK` altogether and rely on the block.json `defaultBlock` value... but I didn't want to assume I could unilaterally make that call. 

So I'm completely open to however you all want this to work this was just my very opinionated approach to respecting the default behavior already present.